cmd/wallet-js-sdk/src/oidc/presentation/openid4vp.js

```diff
@@ -120,6 +129,182 @@ export class OpenID4VP {
         }
       });
 
+    this.client_id = payload.client_id;
+    this.nonce = payload.nonce;
+    this.redirect_uri = payload.redirect_uri;
+
     return response.results;
   }
+
+  /**
+   * submitOIDCPresentation performs an OIDC presentation submission
+   * @param {string} kid - consumer's verification method's kid.
+   * @param {Object} presentationQuery - presentation query object retrieved from user's wallet.
+   * @param {string} issuer - the issuer's key id.
+   * @param {number} expiry - time in seconds representing the expiry of the presentation.
+   *
+   *
+   * @returns {Promise<Object>} - empty promise or error if operation fails.
+   */
+  async submitOIDCPresentation(kid, presentationQuery, issuer, expiry) {
+    if (!kid) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: kid cannot be empty"
+      );
+    } else if (!presentationQuery) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: presentationQuery cannot be empty"
+      );
+    } else if (!presentationQuery.presentation_submission) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: presentation_submission is missing"
+      );
+    } else if (!presentationQuery.type) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: type is missing"
+      );
+    } else if (!presentationQuery.verifiableCredential) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: verifiableCredential is missing"
+      );
+    } else if (!issuer) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: issuer is missing"
+      );
+    } else if (!expiry) {
+      throw new TypeError(
+        "Error submitting OpenID4VP presentation: expiry is missing"
+      );
+    }
+
+    const header = new Object();
+    Object.defineProperty(header, "alg", {
+      value: alg,
+    });
+    Object.defineProperty(header, "kid", {
+      value: issuer,
+    });
+    Object.defineProperty(header, "typ", {
+      value: "JWT",
+    });
+
+    const idToken = await generateIdToken(
+      presentationQuery.presentation_submission,
+      header,
+      expiry
+    );
+
+    const vp = new Object();
+    Object.defineProperty(payload, "@context", {
+      value: presentationQuery["@context"],
+    });
+    Object.defineProperty(payload, "type", {
+      value: presentationQuery.type,
+    });
+    Object.defineProperty(payload, "verifiableCredential", {
+      value: presentationQuery.verifiableCredential,
```

Comment:
'**base64urlencoded**' verifiable credential

Reply:
Right, maybe we could open an issue for it in aries? This shouldn't require any changes here once aries code is updated

Reply:
why we need to have issue in aries, you are directly creating presentation here, you are not using aries.

You just have to change line #206

Reply:
it has to be JWT string not 'base64urlencoded'